### PR TITLE
Sleeping Carp Knee Jerk Nerf: Nyoops Edition

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

I'm still feel like it could do without deflection but we'll keep it for now.

Anyway, sleeping carp users now have distinct weaknesses. Burning them. Lasers, welders, anything that burns. They still have some resistance to it but it's far more mild than their brute mod.

Cooking them is also now not only viable, but encouraged.

Also it's 17TC again.

## Why It's Good For The Game

Watching a miner face off against a sleeping carp user walk menacingly towards them and only just barely staying alive under a mountain of legion cores and medipens while the sleeping carp user is barely affected by the variety of powerful gear they brought to the fight, I kinda realized I probably made a god and not a fish man.

I would suggest merging this one pretty quick.

## Changelog
:cl: NecromancerAnne
balance: Sleeping Carp costs 17tc again.
balance: Sleeping Carp users are distinctly worse against sources of burn damage, except the cold.
balance: Sleeping Carp users are extremely vulnerable to being cooked.
/:cl:

### CODE REVIEW

File: code/datums/martial/sleeping_carp.dm [hunk 1]
```diff
@@ -24,21 +24,22 @@
...
 	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
 	var/atk_verb = pick("kick", "chop", "hit", "slam")
-	var/crit_damage = 0
+	var/crit_damage = 0//this is the critical hit damage added to the attack if it rolls, it starts at 0 because it'll be changed when rolled
 	D.visible_message("<span class='danger'>[A] [atk_verb]s [D]!</span>", \
 					"<span class='userdanger'>[A] [atk_verb]s you!</span>", null, null, A)
 	to_chat(A, "<span class='danger'>You [atk_verb] [D]!</span>")
 	if(prob(10))
-		crit_damage += 20
+		crit_damage += 20//here is where the critical damage is adjusted
```
Comment (optimumtact) [inline code/datums/martial/sleeping_carp.dm]: is this a shitpost?

File: code/datums/martial/sleeping_carp.dm [hunk 2]
```diff
@@ -80,38 +83,39 @@
...
-	if(!isturf(A.loc)) //NO MOTHERFLIPPIN MECHS!
-		return BULLET_ACT_HIT
-	if(A.in_throw_mode)
-		A.visible_message("<span class='danger'>[A] effortlessly swats the projectile aside! They can block bullets with their bare hands!</span>", "<span class='userdanger'>You deflect the projectile!</span>")
-		playsound(get_turf(A), pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, TRUE)
-		P.firer = A
-		P.setAngle(rand(0, 360))//SHING
-		return BULLET_ACT_FORCE_PIERCE
-	return BULLET_ACT_HIT
+//datum/martial_art/the_sleeping_carp/on_projectile_hit(mob/living/carbon/human/A, obj/projectile/P, def_zone)
```
Comment (optimumtact) [inline code/datums/martial/sleeping_carp.dm]: what the f

**Comment by ATH1909** [comment]
I get the reasoning for (and approve of) the other changes here, but why'd deflection have to go?

**Comment by Shadowflame909** [comment]
sec only knows how to kill an antag in one way. through lasers @ATH1909 

Thus if burn damage is its weakness and security cant laser it. Then they lose.

File: code/datums/martial/sleeping_carp.dm (hunk 1 above)
Comment (necromanceranne) [reply code/datums/martial/sleeping_carp.dm]: No I'm just...stupid. I'll fix it.

**Comment by necromanceranne** [comment]
@ATH1909 because they can't really be answered by lasers and temp guns and such if they're untouchable by them they just come and munch your face off. That's it really gamer.

I'm sure they'll make do. It's closer to my original vision.

**Comment by Victor239** [comment]
Previously a miner would have been stunlock punched to death so the fact they could even attempt to heal and survive is impressive already. 

This feels just like facing an op with elite hardsuit and CQC now. Sleeping Carp's defining thing was the ranged deflection because after all, you're giving up ranged weapons yourself. I'd rather have less armour and keep in what made SC unique in the first place.

**Comment by necromanceranne** [comment]
Yeah I'm gonna keep in deflection for the time being. hopefully the armor changes are enough as is the price increase.

**Comment by davethwave** [comment]
On another note, the carp man has the ability to use abilities on himself, like that kick that can send you flying. The carp man can do that to propel himself into other people to drop them or fly over tables.

**Comment by Qustinnus** [comment]
emergent gameplay

**Comment by necromanceranne** [comment]
I was using it today it's very funny. But I think I might just make it an actual move and forbid them from crashing wave kicking themselves. Just for sanities sake.

File: code/datums/martial/sleeping_carp.dm [hunk 3]
```diff
@@ -24,21 +24,22 @@
 	return FALSE
 
 /datum/martial_art/the_sleeping_carp/proc/strongPunch(mob/living/carbon/human/A, mob/living/carbon/human/D)
-	var/obj/item/bodypart/affecting = D.get_bodypart(ran_zone(A.zone_selected))
+	var/obj/item/bodypart/affecting = D.get_bodypart(ran_zone(A.zone_selected))//this var is so that the strong punch is always aiming for the body part the user is targeting and not trying to apply to the chest before deviating
```
Comment (spookydonut) [inline code/datums/martial/sleeping_carp.dm]: this is not codedocs

**Comment by Ty-the-Smonk** [comment]
If you remove deflection let them use guns

File: code/datums/martial/sleeping_carp.dm (hunk 3 above)
Comment (necromanceranne) [reply code/datums/martial/sleeping_carp.dm]: So treat me like an idiot, is it the length or am I documenting this incorrectly?

**Comment by ATH1909** [comment]
> I was using it today it's very funny. But I think I might just make it an actual move and forbid them from crashing wave kicking themselves. Just for sanities sake (action button maybe?)

If you do add an actual unique move for self-crashing wave kicking, could you name it "high jump kick" as a Pokemon reference?

**Comment by necromanceranne** [comment]
Well actually I'm just gonna add in tackling when that gets merged.

**Comment by Exavere** [comment]
Is this in reference to your Carp from last night?  That was one brutal murderbone, but don't forget you were using Fists of the North Star too, which drastically buffs the crap out of it.

**Comment by necromanceranne** [comment]
No, it was from several people coming to me about this and also going to terry to witness it myself.

I just wanted at least one go myself on Bagil ;)

**Review by kriskog — APPROVED** [review]

**Pull request merged**